### DISCUSSION TITLE
`register_devices` action does not support the App Store Connect API

### DISCUSSION THREAD

**Discussion by fujikky** [fastlane/fastlane#17395] [Ideas] [answered]
In my workflow, fully automates device registration by calling fastlane's `register_devices` action.
We are also actively migrating to the App Store Connect API because CI can bypass 2FA authentication.

When I tried to register a new device this time, it seemed that the `register_devices` action of fastlane did not yet support the App Store Connect API.
Any plans to support it?

**Comment by rogerluan** [comment, marked as answer]
Hi @fujikky 👋 
As of _right now_ I can't really tell the timeline to add support to ASCAPI, but I'm certain we will. We're prioritizing the actions based on frequency of use so the first actions that received support to the new API were _deliver_, then _match_, etc. We'll get to `register_devices` at some point for sure.

Also, if you're eager to contribute to _fastlane_, this is a relatively easy task that even a beginner could take, since Spaceship (the engine that powers `register_devices`) already has built-in support for the ASCAPI. Any PRs are welcome and will be highly appreciated!

Keep watching _fastlane_ releases to be notified when support to ASCAPI is added to this action 🙌

**Reply by fujikky** [reply]
@rogerluan Thank you for reply!
I was relieved to know that it was planned to be supported.
I'll read the code to see if I can contribute.

**Reply by rogerluan** [reply]
Hey @fujikky  ! 

Josh took care of this [here](https://github.com/fastlane/fastlane/pull/17426) and this was released as part of [_fastlane_ 2.163.0](https://github.com/fastlane/fastlane/releases/tag/2.163.0) 🚀
